rust/chains/hyperlane-cosmos/src/interchain_gas.rs

```diff
@@ -203,10 +205,33 @@ impl Indexer<InterchainGasPayment> for CosmosInterchainGasPaymasterIndexer {
         &self,
         range: RangeInclusive<u32>,
     ) -> ChainResult<Vec<(InterchainGasPayment, LogMeta)>> {
-        let result = self
-            .indexer
-            .get_range_event_logs(range, Self::interchain_gas_payment_parser)
-            .await?;
+        let logs_futures: Vec<_> = range
+            .map(|block_number| {
+                let self_clone = self.clone();
+                tokio::spawn(async move {
+                    self_clone
+                        .indexer
+                        .get_event_log(block_number, Self::interchain_gas_payment_parser)
+                        .await
+                })
+            })
+            .collect();
+
+        // TODO: this can be refactored when we rework indexing, to be part of the block-by-block indexing
+        let result = future::join_all(logs_futures)
```

Comment:
this fn implementation may be prematurely optimized. Do you think we could get rate limited by querying in parallel @tkporter?